### PR TITLE
Varia: testing, build infra and undefined behaviour

### PR DESCRIPTION
**Pull request by robryk**

### CODE REVIEW

**Comment by robryk** [comment]
I've checked that the golden test passes.

File: .travis.yml [hunk 1]
```diff
@@ -12,10 +12,29 @@ matrix:
             - ubuntu-toolchain-r-test
           packages:
             - wget
+            - libjpeg-progs
 
     - os: osx
       env: BUILD_SYSTEM=bazel
 
+    - os: linux
+      dist: trusty
```
Comment (eustas) [inline .travis.yml]: Default dist + sudo: false could speedup box setup.

Comment (robryk) [inline .travis.yml]: Does it work without sudo if I want to install packages? Will check.

Comment (robryk) [inline .travis.yml]: Installing gcc/g++ takes enough time to make using not-trusty not worthwhile.

**Pull request merged**